octoprint_mrbeam/software_update_information.py

```diff
@@ -100,231 +215,244 @@ def switch_software_channel(plugin, channel):
+                        module_tier = moduleconfig[tierversion]
+                        _set_update_config_from_dict(
+                            moduleconfig, module_tier
+                        )  # set tier config from default settings
+                    if tierversion in module:
+                        module_tier = module[tierversion]
+                        _set_update_config_from_dict(
+                            moduleconfig, module_tier
+                        )  # override tier config from tiers set in config_file
+
+                    _set_update_config_from_dict(
+                        moduleconfig, module
+                    )  # set default config from file for module
+
+                    # have to be after the default config from file
+                    if "beamos_date" in module:
+                        beamos_date_config = module["beamos_date"]
+                        for date, beamos_config in beamos_date_config.items():
+                            _logger.debug(
+                                "date compare %s >= %s -> %s",
+                                beamos_date,
+                                datetime.strptime(date, "%Y-%m-%d").date(),
+                                beamos_config,
+                            )
+                            if (
+                                beamos_date
+                                >= datetime.strptime(date, "%Y-%m-%d").date()
+                            ):
+                                if tierversion in beamos_config:
+                                    beamos_config_module_tier = beamos_config[
+                                        tierversion
+                                    ]
+                                    _set_update_config_from_dict(
+                                        beamos_config, beamos_config_module_tier
+                                    )  # override tier config from tiers set in config_file
+                                _set_update_config_from_dict(
+                                    moduleconfig, beamos_config
+                                )
+
+                    if "{tier}" in moduleconfig["branch"]:
+                        moduleconfig["branch"] = moduleconfig["branch"].format(
+                            tier=get_tier_by_id(tier)
+                        )
+
+                    # get version number
+                    current_version = "-"
+                    _logger.debug(
+                        "pip command check %s %s - %s",
+                        module,
+                        moduleconfig,
+                        "pip_command" not in moduleconfig,
+                    )
+                    if (
+                        "global_pip_command" in module
+                        and "pip_command" not in moduleconfig
+                    ):
+                        moduleconfig["pip_command"] = GLOBAL_PIP_COMMAND
+                    if "pip_command" in moduleconfig:
```

Comment:
This is more for readability, but I guess this is could be an `elif` ? It helps with understanding what we are doing.